server/src/migrations/1714513984514-AddLibraryReadOnly.ts

```diff
@@ -0,0 +1,14 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class AddLibraryReadOnly1714513984514 implements MigrationInterface {
+    name = 'AddLibraryReadOnly1714513984514'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`ALTER TABLE "libraries" ADD "isReadOnly" boolean NOT NULL`);
```

Comment:
I think you need to set a default here (probably `true`) for the not null constraint to work.

Reply:
this file is generated, so I don't think I should manually change it? any idea how I'd get it to generate that?

Reply:
and really we should probably default it to true for external libraries and false for upload libraries

Reply:
You can just add `default: true` to the `@Column` decorator options. Doing a conditional default is more complicated; It should be fine to add a query here like `UPDATE libraries SET "isReadOnly" = false WHERE type = 'UPLOAD'`

Reply:
I think the query + adding `isReadOnly: false` [here](https://github.com/immich-app/immich/blob/2cf63eeeab5ced19f1764545d7b1c0e0684dbd82/server/src/cores/user.core.ts#L104) would get the job done.

Reply:
thanks! updated this